### ISSUE TITLE
Breaks in Kubernetes V0.24.8

### ISSUE THREAD

**Issue by Bryce-Soghigian** [joe-elliott/cert-exporter#129]
⨯ release failed after 0.94s error=failed to build for linux_amd64: # github.com/joe-elliott/cert-exporter/src/checkers
../../../go/pkg/mod/github.com/joe-elliott/cert-exporter@v1.4.0/src/checkers/periodicSecretChecker.go:63:57: not enough arguments in call to client.CoreV1().Secrets(p.namespace).List
        have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
        want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
../../../go/pkg/mod/github.com/joe-elliott/cert-exporter@v1.4.0/src/checkers/periodicSecretChecker.go:74:56: not enough arguments in call to client.CoreV1().Secrets(p.namespace).List
        have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
        want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)

**Comment by Bryce-Soghigian** [comment]
Go mod looks like this 

```go.mod
github.com/Azure/go-autorest/autorest v0.11.28
        github.com/joe-elliott/cert-exporter v1.4.0
        github.com/prometheus/client_golang v1.12.1
        github.com/prometheus/common v0.32.1
        github.com/stretchr/testify v1.8.0
        go.goms.io/aks/rp/protos/hcp v0.3.18
        go.uber.org/zap v1.24.0
        gotest.tools v2.2.0+incompatible
)

require (
        github.com/Azure/azure-sdk-for-go/sdk/azcore v1.2.0 // indirect
        github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.2.0 // indirect
        github.com/Azure/azure-sdk-for-go/sdk/internal v1.0.0 // indirect
        github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute v1.0.0 // indirect
        github.com/Azure/go-autorest v14.2.0+incompatible // indirect
        github.com/Azure/go-autorest/autorest/adal v0.9.21 // indirect
        github.com/Azure/go-autorest/autorest/date v0.3.0 // indirect
        github.com/Azure/go-autorest/autorest/to v0.4.0 // indirect
        github.com/Azure/go-autorest/logger v0.2.1 // indirect
        github.com/Azure/go-autorest/tracing v0.6.0 // indirect
        github.com/AzureAD/microsoft-authentication-library-for-go v0.7.0 // indirect
        github.com/Masterminds/semver v1.5.0 // indirect
        github.com/PuerkitoBio/purell v1.1.1 // indirect
        github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578 // indirect
        github.com/beorn7/perks v1.0.1 // indirect
        github.com/blang/semver v3.5.1+incompatible // indirect
        github.com/cespare/xxhash/v2 v2.1.2 // indirect
        github.com/davecgh/go-spew v1.1.1 // indirect
        github.com/emicklei/go-restful/v3 v3.8.0 // indirect
        github.com/etgryphon/stringUp v0.0.0-20121020160746-31534ccd8cac // indirect
        github.com/globalsign/mgo v0.0.0-20181015135952-eeefdecb41b8 // indirect
        github.com/go-logr/logr v1.2.3 // indirect
        github.com/go-openapi/jsonpointer v0.19.5 // indirect
        github.com/go-openapi/jsonreference v0.19.5 // indirect
        github.com/go-openapi/swag v0.19.14 // indirect
        github.com/gofrs/uuid v4.2.0+incompatible // indirect
        github.com/gogo/protobuf v1.3.2 // indirect
        github.com/golang-jwt/jwt/v4 v4.4.2 // indirect
        github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b // indirect
        github.com/golang/protobuf v1.5.2 // indirect
        github.com/google/gnostic v0.5.7-v3refs // indirect
        github.com/google/go-cmp v0.5.9 // indirect
        github.com/google/gofuzz v1.1.0 // indirect
        github.com/google/uuid v1.1.2 // indirect
        github.com/imdario/mergo v0.3.12 // indirect
        github.com/josharian/intern v1.0.0 // indirect
        github.com/json-iterator/go v1.1.12 // indirect
        github.com/kylelemons/godebug v1.1.0 // indirect
        github.com/leebenson/conform v1.2.3-0.20210309070813-53cf4470cad9 // indirect
        github.com/mailru/easyjson v0.7.6 // indirect
        github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
        github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
        github.com/modern-go/reflect2 v1.0.2 // indirect
        github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect
        github.com/pkg/browser v0.0.0-20210115035449-ce105d075bb4 // indirect
        github.com/pkg/errors v0.9.1 // indirect
        github.com/pmezard/go-difflib v1.0.0 // indirect
        github.com/prometheus/client_model v0.2.0 // indirect
        github.com/prometheus/procfs v0.7.3 // indirect
        github.com/samkreter/redact v0.0.2 // indirect
        github.com/spf13/pflag v1.0.5 // indirect
        go.goms.io/aks/rp/protos/utils v0.0.3 // indirect
        go.goms.io/aks/rp/toolkit/constvalues/gallery v0.0.1 // indirect
        go.uber.org/atomic v1.9.0 // indirect
        go.uber.org/goleak v1.2.0 // indirect
        go.uber.org/multierr v1.7.0 // indirect
        golang.org/x/crypto v0.0.0-20220722155217-630584e8d5aa // indirect
        golang.org/x/mod v0.6.0-dev.0.20220419223038-86c51ed26bb4 // indirect
        golang.org/x/net v0.3.1-0.20221206200815-1e63c2f08a10 // indirect
        golang.org/x/oauth2 v0.0.0-20220223155221-ee480838109b // indirect
        golang.org/x/sys v0.3.0 // indirect
        golang.org/x/term v0.3.0 // indirect
        golang.org/x/text v0.5.0 // indirect
        golang.org/x/time v0.0.0-20220210224613-90d013bbcef8 // indirect
        google.golang.org/appengine v1.6.7 // indirect
        google.golang.org/protobuf v1.28.1 // indirect
        gopkg.in/inf.v0 v0.9.1 // indirect
        gopkg.in/yaml.v2 v2.4.0 // indirect
        gopkg.in/yaml.v3 v3.0.1 // indirect
        k8s.io/api v0.26.0 // indirect
        k8s.io/apimachinery v0.26.0 // indirect
        k8s.io/client-go v11.0.0+incompatible // indirect
        k8s.io/klog/v2 v2.80.1 // indirect
        k8s.io/kube-openapi v0.0.0-20221012153701-172d655c2280 // indirect
        k8s.io/utils v0.0.0-20221107191617-1a15be271d1d // indirect
        sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
        sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
        sigs.k8s.io/yaml v1.3.0 // indirect
)

replace (
        k8s.io/api => k8s.io/api v0.24.8
        k8s.io/apimachinery => k8s.io/apimachinery v0.24.8
        k8s.io/client-go => k8s.io/client-go v0.24.8
)

**Comment by joe-elliott** [comment]
Fixed in #130
